src/app/services/reducers/snippet-reducer.ts

```diff
@@ -1,10 +1,31 @@
 import { IAction } from '../../../types/action';
-import { GET_SNIPPET_SUCCESS } from '../redux-constants';
+import { GET_SNIPPET_ERROR, GET_SNIPPET_PENDING, GET_SNIPPET_SUCCESS } from '../redux-constants';
 
-export function snippets(state = {}, action: IAction): any {
+const initialState = {
+    pending: false,
+    data: {},
+    error: null
+  };
+export function snippets(state = initialState, action: IAction): any {
```

Comment:
>any [](start = 65, length = 3)

great place to use typescript to ensure the type coming out is the state type.

Reply:
I can create a base type for all the initial states that involve fetching data from APIs other than graph can inherit from so that it can be consistent,

This is some chunk of work which will make this PR a headache to review. I can add this to the backlog as technical debt...